rust/src/meas/randomized_response/mod.tex

```diff
@@ -0,0 +1,250 @@
+
+\title{Privacy Proofs for OpenDP: Binary Randomized Response Measurement}
+\author{Vicki Xu, Hanwen Zhang, Zachary Ratliff}
+\date{Summer 2022}
+\begin{document}
+
+\maketitle
+\tableofcontents
+
+%\section{Measurement Description}
+%In binary randomized response, the participant lies with a probability $< \frac{1}{2}$ and tells the truth for the rest of the time. 
+%In Python and in Rust, when booleans are cast to integers, $1$ represents true, $0$ false. 
+%In the function defined line \ref{line:fn}, the true value, \emph{arg}, is taken as input. A boolean is generated in line \ref{line:coinflip} using \texttt{sample\_bernoulli}, which takes in \emph{prob} and \emph{constant\_time} as inputs. As we see from part (a), this means a value of $1$ is generated \emph{prob} percent of the time, and $0$ is generated $1 -$\emph{prob} percent of the time. This is the ``coinflip" step, wherein the computer decides whether to return the true value or the random response. So with probability \emph{prob}, the true value, \emph{arg}, is returned. With probability $1 -$\emph{prob}, the lie, which in the case of the boolean is the opposite (hence $1 - $\emph{arg}), is returned. 
+
+\section{Algorithm Implementation}
+\subsection{Code in Rust}
+
+The current OpenDP library contains the \texttt{make\_randomized\_response\_bool} function implementing the binary randomized response measurement. This is defined in lines 28-53 of the file \texttt{mod.rs} in the Git repository \url{https://github.com/opendp/opendp/blob/main/rust/src/meas/randomized_response/mod.rs#L28-L53}
+
+In \texttt{make\_randomized\_response\_bool}, which accepts a parameter \texttt{prob} of type \texttt{Q} and a parameter \texttt{constant\_time} of type \texttt{bool}, the function takes in a boolean data point \texttt{arg} and returns the truthful value \texttt{arg} with probability \texttt{prob} and the untruthful value $\texttt{!arg}$ with probability $1 - \texttt{prob}$ (in constant time, if the flag is turned on). 
+
+\subsection{Pseudo Code in Python}
+We present a simplified Python-like pseudocode of the Rust implementation below. The necessary definitions for the pseudocode can be found at \href{https://github.com/opendp/whitepapers/blob/pseudocode-defns/pseudocode-defns/pseudocode_defns.pdf}{``List of definitions used in the pseudocode"}. % need to fix
+
+\emph{The use of \texttt{code}-style parameters in the preconditions section below (for example, \texttt{input\_domain}) means that this information should be passed along to the \texttt{Measurements} constructor.}
+
+\subsubsection*{Preconditions}
+To ensure the correctness of the output, we require the following preconditions:
+
+\begin{itemize}
+    \item \textbf{User-specified types:}
+    \begin{itemize}
+        \item Variable \texttt{prob} must be of type \texttt{Q}
+        \item Variable \texttt{constant\_time} must be of type \texttt{bool}
+        \item Type \texttt{bool} must have trait \texttt{SampleBernoulli<Q>}
```

Comment:
I didn't find the def of the trait SampleBernoulli<Q> in the pseudocode defs doc (though I can guess what it means)